integrations/opensearch/src/haystack_integrations/document_stores/opensearch/document_store.py

```diff
@@ -106,9 +125,14 @@ def _get_default_mappings(self) -> Dict[str, Any]:
     @property
     def client(self) -> OpenSearch:
         if not self._client:
-            self._client = OpenSearch(self._hosts, **self._kwargs)
-            # Check client connection, this will raise if not connected
-            self._client.info()  # type:ignore
```

Comment:
Any reason to remove this? 🤔

Reply:
Yes, if you scope OpenSearch calls to certain index patterns using AWS IAM policies, this would break as the resulting request is a "GET /" request against the cluster without any index pattern. Also this breaks AWS OpenSearch Serverless as "GET /" requests are not allowed. 
We don't need it anyways. The following exists() call would raise in a similar way if the cluster is not reachable.

Reply:
Ok, cool. 👍